### PR TITLE
[Front end] Checks to see if user is already logged in before displaying login button

### PR DESCRIPTION
**Pull request by almagee16**

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -69,6 +72,35 @@ const onClickP1 = (
...
+          const profile = googleUser.getBasicProfile();
+          store.dispatch(loginSuccess(profile.getName(), profile.getName(), profile.getEmail())); 
+        }
+      }
+    );
+
+    resolve();
+  });
+});
+
```
Comment (rgreen312) [inline client/src/App.tsx]: This logic shouldn't go in `App.tsx`. Only really high level code should go here (like the routing container). Can we move this into a new action in the `login/actions.ts` file? Then we'll call that action in the `componentDidMount` function the header component.

File: client/src/login/actions.ts [hunk 2]
```diff
@@ -23,24 +24,6 @@ export type LoginError = ReturnType<typeof loginError>;
...
-       * Retrieve the singleton for the GoogleAuth library and set up the
-       * client.
-       */
-      auth2 = gapi.auth2.init({
-          client_id: '634069824484-ch6gklc2fevg9852aohe6sv2ctq7icbk.apps.googleusercontent.com'
-      });
-      resolve();
-    });
-  });
-
```
Comment (rgreen312) [inline client/src/login/actions.ts]: This seems like a good place for this logic to live... Why are you removing it?

File: client/src/App.tsx (hunk 1 above)
Comment (rgreen312) [reply client/src/App.tsx]: Well the UI will display as soon as the page is loaded. We could do the initialization from the `componentDidMount` function in `App.tsx`, but we should do it by calling an action, not by putting all the code for it there

File: client/src/login/actions.ts (hunk 2 above)
Comment (rgreen312) [reply client/src/login/actions.ts]: Yeah, I think the best option is probably to have 2 different actions, one to attempt to login in the background, and one that will only happen when the user clicks the login button

Comment (almagee16) [reply client/src/login/actions.ts]: Ah, okay. Well I don't know how to do that. So we'll talk on Wednesday

**Comment by almagee16** [comment]
This looks good. Can you add a screen recording to show that the refresh functionality still works? And can you also record the signout working?

I'll stamp once you do that!

**Pull request merged**